testsuite/tests/lib-random/testvectors.ml

```diff
@@ -0,0 +1,17 @@
+(* TEST
+*)
+
+(* Check the numbers drawn from a known state against the numbers
+   obtained from the reference Java implementation. *)
+
+open Bigarray
+
+let _ =
+  let a = Array1.of_array Int64 C_layout [| 1L; 2L; 3L; 4L |] in
+  (* Violate abstraction of type Random.State.t to manipulate state directly *)
```

Comment:
I wondered if we should provide a function to set the "raw" state in the library.

Pros:
- no Obj.magic here
- there might be cases where users want to avoid paying the cost of the additional mixing operations

Cons:
- this breaks the generator if an all-zero array is passed
- we have to leak details about the generator algorithm to explain this (which raw seeds to avoid)

In the end I think the current approach is sensible, and no change is required.